src/biz/collab/ops.rs

```diff
@@ -521,6 +635,86 @@ pub async fn get_database_fields(
   Ok(acc)
 }
 
+// inserts a new field into the database
+// returns the id of the field created
+pub async fn add_database_field(
+  uid: i64,
+  collab_storage: &CollabAccessControlStorage,
+  pg_pool: &PgPool,
+  workspace_id: &str,
+  database_id: &str,
+  insert_field: AFInsertDatabaseField,
+) -> Result<String, AppError> {
+  let (mut db_collab, db_body) =
+    get_database_body(collab_storage, workspace_id, database_id).await?;
+
+  let new_id = uuid::Uuid::new_v4()
+    .to_string()
+    .chars()
+    .take(6)
+    .collect::<String>();
+
+  let mut type_options = TypeOptions::new();
+  let type_option_data = insert_field
+    .type_option_data
+    .unwrap_or(serde_json::json!({}));
+  match serde_json::from_value(type_option_data) {
+    Ok(tod) => type_options.insert(insert_field.field_type.to_string(), tod),
+    Err(err) => {
+      return Err(AppError::InvalidRequest(format!(
+        "Failed to parse type option: {:?}",
+        err
+      )));
+    },
+  };
+
+  let new_field = Field {
+    id: new_id.clone(),
+    name: insert_field.name,
+    field_type: insert_field.field_type,
+    type_options,
+    ..Default::default()
+  };
+
+  let db_collab_update = {
+    let mut yrs_txn = db_collab.transact_mut();
```

Comment:
lots of things you do it manually were already implement. For example create a field and update its views.
<img width="966" alt="image" src="https://github.com/user-attachments/assets/8c132642-7e61-4896-b937-348b7ec21a7f">


If we can't use Database directly, maybe we can do some refactor in collab-repo.

Reply:
I've used this